experimental/dds/tree2/src/util/utils.ts

```diff
@@ -265,3 +270,22 @@ export function assertNonNegativeSafeInteger(index: number) {
  * @alpha
  */
 export type Assume<TInput, TAssumeToBe> = TInput extends TAssumeToBe ? TInput : TAssumeToBe;
+
+/**
+ * Only use this function for testing purposes.
+ */
+let count: number | undefined;
+export function useDeterministicStableId(f: () => void) {
+	count = 1;
```

Comment:
i think i'd prefer escape-hatching a "test-only force UUID generation implementation" which we use to pass in v5 uuids in tests (unless they fail stableId assertions b/c we force v4 guids in some places...?). of course ideally this would be properly dependency injected, but it's not too bad of a kludge.

Reply:
We do currently require v4 UUIDs. Session IDs must be v4s, and while the commit tags don't currently need to be v4s, they will need to be v4s if we ever want to compress them with the IdCompressor.

I agree injection would be nice, but it seemed too much for this PR. Ideally I'd like to inject an "IRandom" into the tree at the top level (so anything random can be controlled, not just UUIDs). It's further complicated by the fact that we use the `generateStableId` function from the runtime, it's not a function we own anymore.

Reply:
yeah. I noticed that--it's in the runtime that things should be injectable. But not really a huge deal (or at least, I'm fine with cleaning it up later).

Seems like a smell to me if things other than the id compressor cares that the ids are v4 and not just unique. I don't think we care much to test any v4 uuid properties in snapshot testing.

Reply:
It would be really cool if we had a central RNG for all of Fluid that was easy to mock with an `itDeterministic` or something, and all UUID generation used it.

It's not that anything besides the IdCompressor cares the IDs are v4s, it's that because the IdCompressor cares and because we want to potentially use the IdCompressor to compress these things later, we are ensuring that they are a compatible type for when that happens. If we make them v5s now, and we decide to compress them later, we'll have to make them into v4s again.